src/LEGO.AsyncAPI/Models/AsyncApiOperation.cs

```diff
@@ -81,7 +84,10 @@ public virtual void SerializeV2(IAsyncApiWriter writer)
             }
 
             writer.WriteStartObject();
-            // writer.WriteOptionalProperty(AsyncApiConstants.OperationId, this.Title);
+
+            var operationId = writer.Workspace.RootDocument.Operations.FirstOrDefault(pair => pair.Value == this).Key;
```

Comment:
This doesnt look right. You cant just pick the first operation in the list.
Operations for v2 are serialized not from the root document but from the channels themselves. So this doesnt make sense.

Reply:
at the very least you have a possible null ref as the operation might be null.. if its in components..
I dont think this is a good idea - as operationId cannot be set at all from V3, we shouldn't try to infer it.